lib/core/di/di_utils_module.dart

```diff
@@ -11,10 +12,21 @@ class UtilsDiModule {
   void setupModule(GetIt locator) {
     locator._setupModule();
   }
+
+  Future<void> setupProviders(GetIt locator) async {
+    locator._setupProviders();
+    await locator.allReady();
```

Comment:
its ok to call `allReady` two times?

Reply:
Yes, I think it's ok. 
the providers are the only thing that they are async, so you need the providers to initialize the other things.

The order is:
1. providers 
2. other things

Reply:
> but if they are async and the other are async shouldn't that be fixed with just one allReady?

The issue is that for example you cannot initialize the localsource if you didn't initialize the DB

So, as the DB initialization is async you have to call

await dbInit
await localSource

in this example is like>

await providers
await other things (repositories, sources, utiles, etc)